client/web/web.go

```diff
@@ -65,13 +65,43 @@ type Server struct {
 	browserSessions sync.Map
 }
 
+// ServerMode specifies the mode of a running web.Server.
+type ServerMode string
+
+const (
+	// LoginServerMode serves a readonly login client for logging
+	// a node into a tailnet, and viewing a readonly interface of
+	// the node's current Tailscale settings.
+	//
+	// In this mode, API calls are authenticated via platform auth.
+	LoginServerMode ServerMode = "login"
+
+	// ManageServerMode serves a management client for editing
+	// tailscale settings of a node.
+	//
+	// This mode is restricts the app to only being assessible
+	// over Tailscale, and API calls are authenticated via browser
+	// sessions associated with the source's Tailscale identity.
+	ManageServerMode ServerMode = "manage"
+
+	// LegacyServerMode serves the legacy web client, visible to
+	// users prior to release of tailscale/corp#14335.
+	LegacyServerMode ServerMode = "legacy"
+)
+
 var (
 	exitNodeRouteV4 = netip.MustParsePrefix("0.0.0.0/0")
 	exitNodeRouteV6 = netip.MustParsePrefix("::/0")
 )
 
 // ServerOpts contains options for constructing a new Server.
 type ServerOpts struct {
+	// Mode specifies the mode of web client being constructed.
+	Mode ServerMode
+
+	// DevMode indicates that the server should be started with frontend
+	// assets served by a Vite dev server, allowing for local development
+	// on the web client frontend.
 	DevMode bool
```

Comment:
as just discussed on call, we'll move DevMode to a DEBUG env var

Reply:
Okay with that as a follow-up?

Reply:
sure